plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -574,34 +544,38 @@ public static ColumnMapping oracleTimestampWithTimeZoneColumnMapping()
 
     public static LongWriteFunction oracleTimestampWithTimeZoneWriteFunction()
     {
-        return (statement, index, encodedTimeWithZone) -> {
+        return LongWriteFunction.of(OracleTypes.TIMESTAMPTZ, (statement, index, encodedTimeWithZone) -> {
             Instant time = Instant.ofEpochMilli(unpackMillisUtc(encodedTimeWithZone));
             ZoneId zone = ZoneId.of(unpackZoneKey(encodedTimeWithZone).getId());
             statement.setObject(index, time.atZone(zone));
-        };
+        });
     }
 
     private static WriteMapping oracleBooleanWriteMapping()
     {
-        return WriteMapping.booleanMapping("number(1)", (statement, index, value) -> {
-            statement.setInt(index, value ? 1 : 0);
-        });
+        return WriteMapping.booleanMapping("number(1)", oracleBooleanWriteFunction());
+    }
+
+    private static BooleanWriteFunction oracleBooleanWriteFunction()
+    {
+        return BooleanWriteFunction.of(Types.BOOLEAN, (statement, index, value) -> statement.setInt(index, value ? 1 : 0));
```

Comment:
We map boolean to `number(1)`. 
Why is null typed as `Types.BOOLEAN`? Document.

Reply:
In Oracle you have a bunch of classes that are responsible for binding null values.

There is this internal mapping between SQL type and Oracle type in the `getInternalType`. This method (and other that returns NullBinder) throws on unknown/unhandled type.

Boolean is mapped to VarnumNullBinder. Also BOOLEAN is known type in the `OracleTypes`

Reply:
Thanks for digging this down. Still, we emit numbers (`number(1)`), so why not use `Types.TINYINT` or something like that? 
Or, simply revert the change here. The target column is _not_ of type boolean, so there still will be a cast on Oracle side, right?

Reply:
No. There are a couple of "null binders" in the oracle implementation. Most numeric types use a single null binder with the same behaviour for every type. It doesn't actually matter in that regard (this won't invalidate a cursor)

Reply:
This requires a code comment.  

(IMO if we go `TINYINT` or some other integer type, this wouldn't require a code comment)

Reply:
(unresolved, i don't see resolution)